File: apps/server/src/main/java/com/zl/mjga/platform/events/VersionedDomainEvent.java
package com.zl.mjga.platform.events;

import java.time.Instant;
import java.util.Optional;
import java.util.UUID;

/** Stable event envelope; payload implementations must also have a versioned JSON schema. */
public interface VersionedDomainEvent {
  UUID eventId();

  String eventType();

  int schemaVersion();

  Instant occurredAt();

  String sourceModule();

  String correlationId();

  Optional<String> causationId();
}

File: apps/server/src/main/kotlin/com/zl/mjga/platform/events/VersionedDomainEvent.kt
package com.zl.mjga.platform.events

import java.time.Instant
import java.util.UUID

/** Stable event envelope; payload implementations must also have a versioned JSON schema. */
interface VersionedDomainEvent {
    fun eventId(): UUID

    fun eventType(): String

    fun schemaVersion(): Int

    fun occurredAt(): Instant

    fun sourceModule(): String

    fun correlationId(): String

    fun causationId(): String?
}
